src/components/ArticlesByTopic.js

```diff
@@ -1,24 +1,26 @@
 import { useEffect, useState } from "react";
 import { FetchAllArticlesByTopic } from "./api-logic";
-import { Link } from "react-router-dom";
+import { Link, useSearchParams } from "react-router-dom";
 
-const ArticlesByTopic = () => {
+const ArticlesByTopic = ({ topic }) => {
   const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const topicQuery = searchParams.get("topic");
 
   useEffect(() => {
     setIsLoading(true);
-    FetchAllArticlesByTopic().then((body) => {
+    FetchAllArticlesByTopic(topicQuery, topic).then((body) => {
```

Comment:
The url for this component doesn't work when you refresh, suspect becasue you try to use topic variable to send api response which is undefined when this component isnt navigated to from home. topicQuery may be useful enough on its own.